lib/practice_exercises.rb

```diff
@@ -1,13 +1,42 @@
+# Time Complexity: linear
+# Space Complexity: linear
 
-# Time Complexity: ?
-# Space Complexity: ?
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  # remove white spaces
+  string = string.gsub(/\s+/, "").downcase
+  # reverse string
+  reversed = ""
+  i = string.length - 1
+  while i > -1
+    reversed += string[i]
+    i -= 1
+  end
+  return reversed == string
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
-def longest_prefix(strings)
-  raise NotImplementedError, "Not implemented yet"
-end
 
+# Time Complexity: linear
+# Space Complexity: linear
+def longest_prefix(strings)
```

Comment:
Notice the nested loops here.  I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍